src/main/java/org/jenkinsci/plugins/workflow/cps/EnvActionImpl.java

```diff
@@ -88,7 +96,12 @@ private EnvActionImpl() {
     @Override public String getProperty(String propertyName) {
         try {
             CpsThread t = CpsThread.current();
-            return EnvironmentExpander.getEffectiveEnvironment(getEnvironment(), t.getContextVariable(EnvVars.class), t.getContextVariable(EnvironmentExpander.class)).get(propertyName);
+            TaskListener listener = getListener();
+
+            return EnvironmentExpander.getEffectiveEnvironment(
+                    getEnvironment(getListener()), t.getContextVariable(EnvVars.class),
+                    t.getContextVariable(EnvironmentExpander.class), null, listener)
```

Comment:
Note that if you follow my suggestion in the API PR, this call will then have identical behavior to what it did before—just avoiding a deprecation warning.

(I see no reason for `env.SOMETHING` to behave differently. There is no particular `StepContext` when you are in the middle of Groovy code. There are contexts for enclosing blocks, but the API is defined to allow setting an environment for a step, not for a chain of blocks.)

Reply:
@jglick Ok. Personally I only need the environment inside the steps. I assumed you wanted to keep the behaviour consistent. Let's leave it like that then.